### PR TITLE
Structural adjustment for proxima

### PR DESCRIPTION
**Pull request by rg070836rg**
## What do these changes do?

Structural adjustment for proxima.

- [x] Structural adjustment.
- [x] Add tests for correctness.
- [x] Add tests for distributed.

## Related issue number

NA

### CODE REVIEW

File: mars/learn/proxima/simple_index/builder.py [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+            ctx[out.key] = path
+        else:
+            # write to external file system
+            fs = get_fs(op.index_path, op.storage_options)
+            filename = f'proxima-{out.index[0]}.index'
+            out_path = f'{op.index_path.rstrip("/")}/{filename}'
+            with fs.open(out_path, 'wb') as out_f:
+                with open(path, 'rb') as in_f:
+                    # 32M
+                    chunk_bytes = 32 * 1024 ** 2
```
Comment (wjsi) [inline mars/learn/proxima/simple_index/builder.py:229]: Can this size be configurable?

File: mars/learn/proxima/core.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    proxima = None
+
+from ... import tensor as mt
+
+if proxima:
+    _type_mapping = {
+        np.dtype(np.float16): proxima.IndexMeta.FT_FP16,
+        np.dtype(np.float32): proxima.IndexMeta.FT_FP32,
+        # seems proxima does not support float64
+        # np.dtype(np.float64): proxima.IndexMeta.FT_FP64,
```
Comment (wjsi) [inline mars/learn/proxima/core.py:28]: Is this a TODO? If not, can we handle np.float64 as float32?

**Review by wjsi — COMMENTED** [review]
Please add license for mars/learn/proxima/simple_index/tests/__init__.py

File: mars/learn/proxima/simple_index/builder.py (hunk 1 above)
Comment (qinxuye) [reply mars/learn/proxima/simple_index/builder.py:229]: No need to configure, only used to reduce potential memory cost.

**Review by wjsi — APPROVED** [review]
LGTM

**Pull request merged**